tests/collections/asr/test_asr_filterbankfeatures_seq_len.py

```diff
@@ -28,56 +26,74 @@ def test_seq_len(self):
+        assert librosa_spec.shape[1] == fb_spec.shape[2], f"{librosa_spec.shape} != {fb_spec.shape}"
 
     @pytest.mark.unit
     def test_random_stft_sizes(self):
         for _ in range(5):
-            nfft = random.randint(128, 2048)
-            window_size = random.randint(128, nfft)
-            hop_size = random.randint(64, window_size)
+            nfft = 2 ** np.random.randint(7, 12)
+            window_size = np.random.randint(100, nfft)
+            hop_size = np.random.randint(64, window_size)
             fb_module = FilterbankFeatures(
-                exact_pad=False, pad_to=1, n_fft=nfft, n_window_size=window_size, n_window_stride=hop_size
+                exact_pad=False,
+                pad_to=1,
+                n_fft=nfft,
+                n_window_size=window_size,
+                n_window_stride=hop_size,
+                normalize=False,
             )
-            audio_length = random.randint(nfft, 2 ** 16)
+            audio_length = np.random.randint(nfft, 2 ** 16)
             test_1 = torch.randn(1, audio_length)
             test_1_len = torch.tensor([audio_length])
-            result, result_len = fb_module(test_1, test_1_len)
+            fb_spec, fb_len = fb_module(test_1, test_1_len)
             assert (
-                result.shape[2] == result_len[0]
-            ), f"{result.shape} != {result_len}: {nfft}, {window_size}, {hop_size}, {audio_length}"
+                fb_spec.shape[2] == fb_len[0]
+            ), f"{fb_spec.shape} != {fb_len}: {nfft}, {window_size}, {hop_size}, {audio_length}"
 
-            spec = librosa.stft(
+            librosa_spec = librosa.stft(
                 test_1.cpu().detach().numpy().squeeze(), n_fft=nfft, hop_length=hop_size, win_length=window_size
             )
 
             assert (
-                spec.shape[1] == result.shape[2]
-            ), f"{result.shape} != {spec.shape}: {nfft}, {window_size}, {hop_size}, {audio_length}"
+                librosa_spec.shape[1] == fb_spec.shape[2]
+            ), f"{librosa_spec.shape} != {fb_spec.shape}: {nfft}, {window_size}, {hop_size}, {audio_length}"
 
+    @pytest.mark.unit
+    def test_random_stft_sizes_exact_pad(self):
         for _ in range(5):
-            nfft = random.randint(128, 2048)
-            window_size = random.randint(128, nfft)
-            hop_size = random.randint(64, window_size)
+            nfft = 2 ** np.random.randint(7, 12)
+            window_size = np.random.randint(100, nfft)
+            hop_size = np.random.randint(64, window_size)
+            if hop_size % 2 == 1:
+                hop_size = hop_size - 1
             fb_module = FilterbankFeatures(
-                exact_pad=True, pad_to=1, n_fft=nfft, n_window_size=window_size, n_window_stride=hop_size
+                exact_pad=True,
+                pad_to=1,
+                n_fft=nfft,
+                n_window_size=window_size,
+                n_window_stride=hop_size,
+                normalize=False,
```

Comment:
This test tends to produce mels with sequence_length = 1, which causes normalize to fail since it does std over time. Just ignore normalization during this test since that's not what it is testing.